client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -185,10 +194,25 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        manageSubscriptions(config);
         Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
         synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
     }
 
+    private void manageSubscriptions(ReaderGroupConfig config) {
+        ReaderGroupConfig oldConfig = getReaderGroupConfig();
+        Set<Stream> oldStreams = oldConfig.isSubscriberForRetention() ? oldConfig.getStartingStreamCuts().keySet() : Collections.emptySet();
+        Set<Stream> newStreams = config.isSubscriberForRetention() ? config.getStartingStreamCuts().keySet() : Collections.emptySet();
+
+        Collection<Stream> streamsToSub = filterOut(newStreams, oldStreams);
+        Collection<Stream> streamsToUnsub = filterOut(oldStreams, newStreams);
+
+        // Unsubscribe to older streams
+        streamsToUnsub.forEach(s -> getThrowingException(controller.deleteSubscriber(scope, s.getStreamName(), groupName)));
```

Comment:
there is an opportunity to perform all these functions in parallel and then combine them in a single get. its just that since we do not have an executor here, all these futures will be executed on forkjoin pool. 

but the code can look as follows:
```
toDeleteFutures = Futures.allof(streamsToUnsub.stream().map(x -> controller.deleteSubscriber(scope, x.getStreamname(), groupName).collect(Collectors.toList()));

toAddFutures = Futures.allof(streamsToSub.stream().map(x -> controller.addSubscriber(scope, x.getStreamname(), groupName).collect(Collectors.toList()));

getThrowingException(CompletableFuture.allof(toAddFutures, toDeleteFutures);
```

```